damage.py

```diff
@@ -21,6 +21,27 @@ def pre_mitigation_damage(
     return (base_damage + bonus_damage) * damage_modifier_percent_mult_factor + damage_modifier_flat
 
 
+def pre_mitigation_spell_damage(
+    base_spell_damage: float,
+    ratio: float,
+    base_damage: float,
+    bonus_damage: float,
+    damage_modifier_flat: float = 0,
+    damage_modifier_percent_mult_factor: float = 1,
+):
+    """
+    Calculates the pre-mitigation damage of a spell or an autoattack
+    All values regarding damage modifiers should include the buffs/debuffs coming from spells, summoner spells, or items
+    from both the attacker AND the defender
+    """
+
+    return (
```

Comment:
```suggestion
    return (base_spell_damage + ratio * (base_damage + bonus_damage)) * damage_modifier_percent_mult_factor + damage_modifier_flat
```
Verified in practice tool with xerath w (increased damage in the middle of the spell) and horizon focus (increased damage over 700 range)

Reply:
est ce que c'est le cas pour les function autoattack aussi ?

Reply:
![image](https://user-images.githubusercontent.com/32098343/204086752-e3069306-fdea-4f41-be07-794dd9fa6f2f.png)

en fait je pense que la le dmg modifier mult c'est bien 1.667, mais faut quand meme laisser le damage flat en dernier en dehors de la parenthese

Reply:
pcq la ca marche juste pcq le damage modif flat =0. Si tu refais les tests avec un damage mod flat non nul je pense que la formule ne marche plus. 
A tester en tt cas

Reply:
ca veut dire que pour les resistances c'est d'abord percent modif puis flat modif 
Et pour les damage modif c'est flat modif puis percent modif ?

Reply:
Oui, l'ordre pour les résistances restent le même que celui qu'on a définit. Et pour les autoattack c'est d'abord flat modif (non affecté par les crit!), et ensuite percent modif. Justement le test avec mf c'est avec autoattack. On peut encore refaire un test mf prowler's claw + des items de crit chance pour encore revérifier la formule